search-provider/eks-discovery-feed-provider.c

```diff
@@ -244,10 +307,46 @@ article_card_descriptions_cb (GObject *source,
       g_variant_builder_open (&builder, G_VARIANT_TYPE ("a{ss}"));
 
       EkncContentObjectModel *model = l->data;
+      guint flags = 0;
+
+      /* Examine the discovery-feed-content object first and set flags
+       * for things that we've overridden */
+      g_autoptr(GVariant) discoveryFeedContentVariant;
+      g_object_get (model,
+                    "discovery-feed-content",
+                    &discoveryFeedContentVariant,
+                    NULL);
+      GVariantIter discoveryFeedContentIter;
+      g_variant_iter_init (&discoveryFeedContentIter,
+                           discoveryFeedContentVariant);
+
+      gchar *key;
+      GVariant *value;
+
+      while (g_variant_iter_loop (&discoveryFeedContentIter, "{sv}", &key, &value))
+        {
+          if (g_strcmp0(key, "blurbs") == 0)
+            {
+              g_autoptr(GPtrArray) array = variant_array_variant_to_ptr_array(value,
```

Comment:
I think you can avoid putting this into a `GPtrArray` altogether, and simply select randomly a child variant from the array using `g_variant_get_child_value ()` or a similar function